src/main/cli/settings.c

```diff
@@ -1223,7 +1223,7 @@ const clivalue_t valueTable[] = {
     { PARAM_NAME_TPA_MODE,          VAR_UINT8  | PROFILE_VALUE | MODE_LOOKUP, .config.lookup = { TABLE_TPA_MODE }, PG_PID_PROFILE, offsetof(pidProfile_t, tpa_mode) },
 #endif
     { PARAM_NAME_TPA_RATE,          VAR_UINT8  | PROFILE_VALUE, .config.minmaxUnsigned = { 0, TPA_MAX}, PG_PID_PROFILE, offsetof(pidProfile_t, tpa_rate) },
-    { PARAM_NAME_TPA_BREAKPOINT,    VAR_UINT16 | PROFILE_VALUE, .config.minmaxUnsigned = { PWM_PULSE_MIN, PWM_PULSE_MAX }, PG_PID_PROFILE, offsetof(pidProfile_t, tpa_breakpoint) },
+    { PARAM_NAME_TPA_BREAKPOINT,    VAR_UINT16 | PROFILE_VALUE, .config.minmaxUnsigned = { 1000, 1990 }, PG_PID_PROFILE, offsetof(pidProfile_t, tpa_breakpoint) },
```

Comment:
The whole point of using a value like PWM_PULSE_MIN, and PWM_PULSE_MAX is to get rid of "magic" numbers. The name they have gets rid of the "magic" that using these arbitrary numbers has. No need for this.

Reply:
It's more complicated here. 1000-2000 is used in later code, PWM values are mixed with integer representation of floating point.

Reply:
If the `tpa_breakpoint` value could be configured to 2000, and it could, there would be a divide by zero error.

And, if less than 1000, we would get a strangely negative value further on in the code, and I don't think that would be intended.
 
Essentially, the values for `PWM_PULSE_MIN` and `PWM_PULSE_MAX` don't have any particular relevance to the throttle range used to calculate TPA, and allow values that we should not permit.  They are as 'magic' as any other value we might use to constrain the range of user inputs.  

It is simpler for the user to have a range starting at 1000 and running up to some value close to 2000.

What would you suggest this range should be?

Reply:
> running up to some value close to 2000.

This is ugly. Just allow value from 1.0 to 2.0 (multiplied by 1000)  and handle upper extreme in the code.

BTW: How does TPA work in 3D mode?

Reply:
TPA settings have always been ugly :-)

The typical user sets a value for the breakpoint between 1000 and 2000, and in general understand that 1000 means zero throttle and 2000 means max throttle.  And that range, for configuring breakpoint, has been like that forever, so I don't want to change it now to something different.

It gets somewhat pointless to diminish P and/or D abruptly, and only, in the top 5% of the throttle range.  There's no virtue in that.  From a practical perspective, setting a breakpoint much above about 1800 is not very sensible; when I looked at this, I was intending to make the upper limit 1800 to prevent people configuring it any higher.  In the end I thought, well, maybe someone will want to try it.. so I chose 1950.  Default is 1350 at present.  

It's the 'attenuation' value always confuses people... :-) 

I have no clue how TPA works in 3D mode.  I expect it wouldn't work at all well :-)